Missing credential button on NodeRed 2.2.2

Hello,

On Node-Red (v2.2.2) I have installed the node-red-contrib-netatmo-homecoach library (v1.0.0) and the node "get home coaches data" does not have the edit button to introduce the credentials in order to connect to netatmo cloud and get the data from the homecoach sensors.

This is very strange because I had installed the same library on Node Red (v1.2.6) in a Rasberry Pi with the same library "node-red-contrib-netatmo-homecoach (v1.0.0)" and it has the button and works properly.

Anybody has experience a similar issue? What can I do to solve this problem?

Missing button on Node Red v2.2.2
![image](https://user-images.githubusercontent.com/74243267/181754450-edf7de09-77d1-4abc-8f7f-f13cacf01ffb.png)

On Node Red v1.2.6 with the button
![image](https://user-images.githubusercontent.com/74243267/181754511-288fa7e9-7551-4812-9b2e-6fcd2893083d.png)

Hello,

I have updated Node-red to v3.0.1 in other PC and it does not let you install the ***@***.******@***.***> library.

Regards,

David

2022-08-01T13:02:42.618Z npm install --no-audit --no-update-notifier --no-fund --save --save-prefix=~ --production --engine-strict ***@***.******@***.***>
2022-08-01T13:02:44.773Z [err] npm
2022-08-01T13:02:44.773Z [err]  ERR! code EBADENGINE
2022-08-01T13:02:44.784Z [err] npm ERR!
2022-08-01T13:02:44.784Z [err] engine Unsupported engine
2022-08-01T13:02:44.784Z [err] npm ERR! engine Not compatible with your version of node/npm: ***@***.******@***.***>
2022-08-01T13:02:44.784Z [err] npm ERR! notsup Not compatible with your version of node/npm: ***@***.******@***.***>
2022-08-01T13:02:44.784Z [err] npm ERR! notsup Required: {"node":">= 14.0.0"}
2022-08-01T13:02:44.784Z [err] npm ERR!
2022-08-01T13:02:44.784Z [err] notsup Actual:   {"npm":"8.5.1","node":"v12.22.9"}
2022-08-01T13:02:44.785Z [err]
2022-08-01T13:02:44.785Z [err] npm
2022-08-01T13:02:44.785Z [err] ERR! A complete log of this run can be found in:
2022-08-01T13:02:44.785Z [err] npm ERR!     /home/los-laqc/.npm/_logs/2022-08-01T13_02_42_980Z-debug-0.log
2022-08-01T13:02:44.801Z rc=1


De: Pierre M ***@***.***>
Enviado el: viernes, 29 de julio de 2022 17:15
Para: devleaks/node-red-contrib-netatmo-homecoach ***@***.***>
CC: David Ramos ***@***.***>; Author ***@***.***>
Asunto: Re: [devleaks/node-red-contrib-netatmo-homecoach] Missing credential button on NodeRed 2.2.2 (Issue #2)


This message was sent from outside of Gestamp. Please do not click links or open attachments unless you recognise the source of this email and know the content is safe.

I have no idea. node-red probably changed the way those "accessory" information are supplied.
I'll have a look in the following days.
Thanks for reporting.

—
Reply to this email directly, view it on GitHub<https://github.com/devleaks/node-red-contrib-netatmo-homecoach/issues/2#issuecomment-1199505432>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ARWNZQ2YCLLHI44AKX2WBKDVWPYPJANCNFSM55ARE2BQ>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

--- a/netatmo_node_get_homecoaches.html
+++ b/netatmo_node_get_homecoaches.html
@@ -19,7 +19,7 @@
         color: '#669966',
         defaults: {
             name: {value:""},
-            creds: {value:"",type:"NetatmoConfigNode"}
+            creds: {value:"",type:"configNode"}
         },
         inputs:1,
         outputs:1,
@@ -32,7 +32,7 @@
 
 <script type="text/x-red" data-template-name="get home coaches data">
     <div class="form-row">
-        <label for="node-input-client_id"><i class="icon-tag"></i> Creds</label>
+        <label for="node-input-creds"><i class="fa fa-server"></i> Creds</label>
         <input type="text" id="node-input-creds" placeholder="Add netatmo creds">
     </div>
     <div class="form-row">
@@ -46,3 +46,4 @@
    <p>Outputs an object called <b>msg</b> containing <b>msg.payload</b> in JSON format (of course).</p>
    <p>See <a href="https://dev.netatmo.com/dev/resources/technical/reference/healthyhomecoach/gethomecoachsdata" target=_blank>https://dev.netatmo.com/dev/resources/technical/reference/healthyhomecoach/gethomecoachsdata</a> for details on format</p>
 </script>
+

--- a/netatmo_node_get_homecoaches.js
+++ b/netatmo_node_get_homecoaches.js
@@ -28,12 +28,13 @@ module.exports = function(RED) {
 
             var dateFormat = require('dateformat');
 
-            var auth = {
-                "client_id": this.creds.client_id,
-                "client_secret": this.creds.client_secret,
-                "username": this.creds.username,
-                "password": this.creds.password
+            const auth = {
+                "client_id": this.creds.credentials.client_id,
+                "client_secret": this.creds.credentials.client_secret,
+                "username": this.creds.credentials.username, 
+                "password": this.creds.credentials.password
             };
+            
             var api = new netatmo(auth);
             api.getHealthyHomeCoachData(function(err, devices) {
                 console.log(dateFormat(new Date(), "dd mmm HH:MM:ss") + " - [info] [node-red-contrib-netatmo-healthyhomecoach] getHealthyHomeCoachData > OK");

